### PR TITLE
min_width parameter for RotatingTextWidget (and music)

### PR DESCRIPTION
**Pull request by adeebshihadeh**
* when min_width < max_width, width conforms to text within bounds of min_width and max_width
* min_width == max_width, width is fixed
* min_width > max_width, min_width acts as a max width

### CODE REVIEW

**Comment by ammgws** [comment]
Seems OK to me, but I haven't had a chance to test it yet. Can you please tell em which of the below this fixes?

- the block changing size between songs
- the block changing size during text rotation
- the block misbehaving with CJK text

**Comment by quillford** [comment]
This only changes the size between songs

**Comment by ammgws** [comment]
Thanks. Hopefully I can merge this soon.

File: src/widgets/rotatingtext.rs [hunk 1]
```diff
@@ -130,14 +134,16 @@ impl RotatingTextWidget {
...
         self.rendered = json!({
-            "full_text": format!("{}{} ",
-                                self.icon.clone().unwrap_or_else(|| String::from(" ")),
-                                self.get_rotated_content()),
+            "full_text": txt,
             "separator": false,
             "separator_block_width": 0,
-            "min_width": if self.content == "" {"".to_string()} else {"0".repeat(self.width+self.icon.clone().unwrap_or_else(|| String::from(" ")).chars().count()+1)},
-            "align": "left",
+            "min_width": if self.content == "" {"".to_string()} else {"0".repeat(max(txt.len(), self.min_width)+1)},
```
Comment (ammgws) [inline src/widgets/rotatingtext.rs:145]: This is a regression for CJK text. Better with `.chars().count()`

**Comment by ammgws** [comment]
I _believe_ #787 covers this. Re-open otherwise.

**Pull request closed without merging**